File: python/BinaryTree.py
class BinaryTree:
    def __init__(self):
        self.root = None
        self.nodes = {}

    @property
    def root(self):
        return self.__root

    @root.setter
    def root(self, root):
        self.__root = root

    @property
    def nodes(self):
        return self.__nodes

    @nodes.setter
    def nodes(self, nodes):
        self.__nodes = nodes

    def __repr__(self):
        return "{" + "root:{root}, nodes:{nodes}".format(root=self.root, nodes=self.nodes) + "}"

File: c#/BinaryTree.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace binaryTree
{
    class BinaryTree
    {
        public string Root { get; set; }
        public Dictionary<string, Node> Nodes {get; set;}

        public BinaryTree()
        {
            this.Root = null;
            this.Nodes = new Dictionary<string, Node>();
        } 


        public override string ToString()
        {
            var nodesList = "";
            foreach (var node in Nodes)
            {
                nodesList += (node.ToString() + "\n");
            }

            return "BinaryTree {" + "root:" + Root + ", nodes:" + nodesList + "}";
        }
    }
}
